### ISSUE TITLE
AltGr + Space combination doesn’t work all of a sudden

### ISSUE THREAD

**Issue by pzehner** [microsoft/terminal#11649]
### Windows Terminal version (or Windows build number)

1.11.*
### Other Software

French Bépo keyboard layout.

### Steps to reproduce

Using the French Bépo keyboard layout, I type the underscore key "_" with AltGr + Space.

### Expected Behavior

Usually it just works.

### Actual Behavior

Today, without any updates or changes in configuration, MS Terminal doesn’t type the character any more. Other AltGr combinations work, but not this one. Any other application (including PowerShell and CMD) can type the character normally.

*Note:* It is more likely that this behaviour started with the 1.11 version. 1.10.2714.0 is the latest behaving correctly.

*Note 2:* "all of a sudden" because MS Terminal updates itself automatically and I could not find a way to prevent that.

**Comment by arthaud** [comment]
Same here.
By the way, I think I know you @pzehner and @nchataing. Bépo users right? :)

**Comment by Siphonay** [comment]
I am also encountering this issue

**Comment by at-github** [comment]
Same as well, i didn’t see shortcut overwriting this combination

**Comment by pzehner** [comment]
I confirm that the problem does not appear by manually downgrading to 1.10.2714.0.

**Comment by crequill** [comment]
I have the same problem with terminal 1.11.2921.0 but it’s corrected in terminal 1.12.2931.0.

**Comment by Geobert** [comment]
Same issue for me, bépo as well. It's very annoying as it's how we type `_` (underscore) in this layout :-/

**Comment by Siphonay** [comment]
Can confirm the issues seems to be fixed in Preview Release 1.12.2931.0

**Comment by chikamichi** [comment]
A few days back, I installed 1.12.2931.0 and it seemed to be fixed. But all of a sudden, in stopped working. No version change.

**Comment by lhecker** [comment]
Hmm indeed this is hard to reproduce... I've downloaded [the portable version](https://download.tuxfamily.org/dvorak/windows/1.1rc2/pkl-bepo-1.1rc2.zip) and AltGr+Space produces "_".

**Comment by Siphonay** [comment]
> Hmm indeed this is hard to reproduce... I've downloaded [the portable version](https://download.tuxfamily.org/dvorak/windows/1.1rc2/pkl-bepo-1.1rc2.zip) and AltGr+Space produces "_".

Be aware that the portable version is a PKL, which are repackaged AutoHotKey scripts, and not a proper keyboard layout per se, so the behavior between this and the actual Windows keyboard layout package may exhibit different behavior

**Comment by pzehner** [comment]
Even downgrading the terminal to the latest 1.10 version and pinning it (I am using Chocolatey), the application automatically updates itself like once a week and I end with the broken 1.11 version…

**Comment by lhecker** [comment]
This line in the 1.11 branch should say `IsLeftAltPressed()` and was introduced in #10988: https://github.com/microsoft/terminal/blob/7b775684e8206abdd12a6720252d0888f1c9d6d0/src/cascadia/TerminalControl/TermControl.cpp#L912

In the 1.12 branch we switched to using regular actions with #11086. Since actions are never run for AltGr key combinations, this  fixes the issue.
For 1.11 I fixed this issue in #11832. The fix isn't particularly beautiful, but I believe it's enough for an older release like that. 😅
